lib/govuk_web_banners/test_validators/recruitment_banner.rb

```diff
@@ -0,0 +1,54 @@
+require "active_support/core_ext/integer/time"
+
+module GovukWebBanners
+  module TestValidators
+    class RecruitmentBanner
+      attr_reader :errors
+
+      def initialize(banners)
+        @errors = {}
+
+        validate(banners)
+      end
+
+      def valid?
+        @errors.keys.none?
+      end
+
+    private
+
+      def add_error(banner, error)
+        @errors[safe_name(banner)] ||= []
+        @errors[safe_name(banner)] << error
+      end
+
+      def safe_name(banner)
+        banner.name || "unnamed banner"
+      end
+
+      def validate(banners)
+        banners.each do |banner|
+          add_error(banner, "is missing a suggestion_text") unless banner.suggestion_text.present?
+          add_error(banner, "is missing a suggestion_link_text") unless banner.suggestion_link_text.present?
+          add_error(banner, "is missing a survey_url") unless banner.survey_url.present?
+          add_error(banner, "is missing any page_paths") unless banner.page_paths.present?
+
+          (banner.page_paths || []).each do |path|
+            add_error(banner, "page_path #{path} should start with a /") unless path.start_with?("/")
```

Comment:
Why would we need to check content-store for draft content? I was thinking we'd just do Services.content_store.content_item("/base_path") and raise an error if it returns anything except a 200. Or am I missing something and it's problematic to do that for some reason?

If you both prefer not to, it's not a blocker from my pov. I'm just thinking it's a pain if we release the gem with a typo. Presume the workflow would be make a branch of the gem, and point frontend apps at it locally to confirm it looks ok, and then open a PR for the branch of the gem. 

But if we have lots of simultaneous banner requests, that workflow becomes time consuming, and this one extra validation feels like it completes the set nicely, so that you could almost just merge the gem bump in confidence that the banners will show.

Reply:
Oh, checking for draft content pages is for pages that will have a banner but aren't live yet. For example if the banner and the page are going live at the same time.

But your point about releasing the gem is a good one. It is much more hassle to release a gem than it is to deploy an application to correct a typo.